deploy/charts/approver-policy/templates/metrics-service.yaml

```diff
@@ -5,7 +5,7 @@ metadata:
   name: {{ include "cert-manager-approver-policy.name" . }}-metrics
   namespace: {{ .Release.Namespace | quote }}
   labels:
-    app: {{ include "cert-manager-approver-policy.name" . }}
+    app: {{ include "cert-manager-approver-policy.name" . }}-metrics
```

Comment:
The disadvantage of this, is that I can no longer easily discover all the resources related to this `app`. E.g.

```bash
$ kubectl get all --namespace cert-manager --selector app=cer
t-manager-approver-policy
NAME                                               READY   STATUS    RESTARTS   AGE
pod/cert-manager-approver-policy-c68dc44c9-5785f   1/1     Running   0          107m

NAME                                           TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
service/cert-manager-approver-policy           ClusterIP   10.96.10.146    <none>        443/TCP    107m
service/cert-manager-approver-policy-metrics   ClusterIP   10.96.133.139   <none>        9402/TCP   107m

NAME                                                     DESIRED   CURRENT   READY   AGE
replicaset.apps/cert-manager-approver-policy-c68dc44c9   1         1         1       107m
```

Reply:
@wallrj

I was looking at the same GitHub issue, and indeed you are right: targetPort setup the Prometheus's scrape config to scrape the container directly (for each service discovered, hence the duplicated target).
So 2 solutions: tightening the service discovery (my first approach), or targeting the service (by name) directly and not the container (your approach).

Happy to implement your approach, I like to think of a ServiceMonitor to actually be setup with service attributes and not container ones...

Reply:
Great. Then please modify the PR to use the the `port: metrics` approach and ping me for another review.